src/backend/commands/matview.c

```diff
@@ -4338,3 +4349,47 @@ isIvmColumn(const char *s)
 {
 	return (strncmp(s, "__ivm_", 6) == 0);
 }
+
+/*
+ * get_securityQuals
+ *
+ * Get row security policy on a relation.
+ * This is used by IVM for copying RLS from base table to enr.
+ */
+static List *
+get_securityQuals(Oid relId,Query *query)
+{
+	ParseState *pstate;
+	Relation rel;
+	ParseNamespaceItem *nsitem;
+	RangeTblEntry *rte;
+	List *securityQuals;
+	List *withCheckOptions;
+	bool  hasRowSecurity;
+	bool  hasSubLinks;
+
+	securityQuals = NIL;
+	pstate = make_parsestate(NULL);
+
+	rel = table_open(relId, NoLock);
+	nsitem = addRangeTableEntryForRelation(pstate, rel, AccessShareLock, NULL, false, false);
+	rte = nsitem->p_rte;
+
+	get_row_security_policies(query, rte, 1,
```

Comment:
I would like an explanation for why the third argument is 1. Also, I think some comments is necessary for the explanation.  In my understanding, this is because all ENRs are accessed in a subquery which has only one RTE and rtindex is always 1, right?

Reply:
What you say is correct.
Currently, get_securityQuals function is used by IVM only if rt_index is 1.
Althouth it is no problem to set rt_index to 1, it is dengerous to use fixed value. I would like to add new argument rt_index on get_securityQuals().